pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Consumer.java

```diff
@@ -776,6 +863,12 @@ public void redeliverUnacknowledgedMessages(List<MessageIdData> messageIds) {
             LongPair longPair = pendingAcks.get(position.getLedgerId(), position.getEntryId());
             if (longPair != null) {
                 long batchSize = longPair.first;
+                if (isDeletionAtBatchIndexLevelEnabled()) {
+                    long[] cursorAckSet = getCursorAckSet(position);
+                    if (cursorAckSet != null) {
+                        batchSize -= BitSet.valueOf(cursorAckSet).cardinality();
```

Comment:
The `cardinality` of the `cursorAckSet` means un-acked message count, right?

Reply:
batchSize -= BitSet.valueOf(cursorAckSet).cardinality() is equals with `batchSize = batchSize - BitSet.valueOf(cursorAckSet).cardinality()`.
And `batchSize - BitSet.valueOf(cursorAckSet).cardinality()` means how many un-ack msg count.

Reply:
> The `cardinality` of the `cursorAckSet` means un-acked message count, right?

Yes, right. make a mistake here.